### PR TITLE
Update pd csi driver images to use v1 images (CSINode, CSIDriver, etc)

### LINKED ISSUE #102077: e2e failures: all pdcsi tests
#### Which jobs are failing:
ci-kubernetes-e2e-gci-gce-serial

#### Which test(s) are failing:
All gce pd csi tests

#### Since when has it been failing:
5/14

#### Testgrid link:
https://k8s-testgrid.appspot.com/sig-storage-kubernetes#gce-serial

#### Reason for failure:
Likely related to the removal of v1beta1 storage objects in https://github.com/kubernetes/kubernetes/pull/99840. The provisioner and attacher sidecars need to be updated.

### PR DESCRIPTION
**Pull request by mattcary**
/kind bug
/kind failing-test
/sig storage
Fixes #102077

```release-note
None
```

/assign @msau42

### CODE REVIEW

**Comment by mattcary** [comment]
@msau42 

I noticed the controller gce pd driver was at v1.0.1 and the node was at v0.7.0, any reason for the skew or is that just an oversight?

At any rate these manifests will soon go away as we are going to be installing the driver by default in kube-up.

**Comment by msau42** [comment]
> I noticed the controller gce pd driver was at v1.0.1 and the node was at v0.7.0, any reason for the skew or is that just an oversight?

I think it was just oversight

File: test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml [hunk 1]
```diff
@@ -51,12 +106,32 @@ spec:
...
+              port: http-endpoint
+            initialDelaySeconds: 10
+            timeoutSeconds: 10
+            periodSeconds: 20
           volumeMounts:
             - name: socket-dir
               mountPath: /csi
         - name: gce-pd-driver
-          image: gcr.io/gke-release/gcp-compute-persistent-disk-csi-driver:v1.0.1-gke.0
+          image: gcr.io/gke-release/gcp-compute-persistent-disk-csi-driver:v1.2.1-gke.0
```
Comment (msau42) [inline test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml:134]: We haven't fixed the k8s image building issue yet?

Comment (mattcary) [reply test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml:134]: Thanks, I just copied this from stable-master on the upstream driver. But that's out of date, we've moved to v1.2.2-gke.2 on GKE already.

**Comment by mattcary** [comment]
I missed the leader election rbac.

**Comment by mattcary** [comment]
Darn it, 1.2.2 is still in staging. Switching back to 1.2.1 until I figure
out when the promotion will happen.

**Comment by mattcary** [comment]
Oh, wait, we should be using the upstream cloud build jobs now.

File: test/utils/image/manifest.go [hunk 2]
```diff
@@ -89,19 +90,20 @@ func initReg() RegistryList {
...
+		E2eRegistry:              "gcr.io/kubernetes-e2e-test-images",
+		PromoterE2eRegistry:      "k8s.gcr.io/e2e-test-images",
+		BuildImageRegistry:       "k8s.gcr.io/build-image",
+		InvalidRegistry:          "invalid.com/invalid",
+		GcEtcdRegistry:           "k8s.gcr.io",
+		GcRegistry:               "k8s.gcr.io",
+		SigStorageRegistry:       "k8s.gcr.io/sig-storage",
+		PrivateRegistry:          "gcr.io/k8s-authenticated-test",
+		SampleRegistry:           "gcr.io/google-samples",
+		GcrReleaseRegistry:       "gcr.io/gke-release",
```
Comment (msau42) [inline test/utils/image/manifest.go:103]: I think we can actually remove this one too. But let's leave it to a separate PR

Comment (mattcary) [reply test/utils/image/manifest.go:103]: +1 thanks. I meant to ask if I was doing this manifest stuff correctly, I don't understand how it works, I was just adding stuff until the build worked.
